### PR TITLE
eslint: Indent and newline-per-chained-call rules

### PR DESCRIPTION
**Pull request by mareklibra**
Rule enabled: `indent` and set to 4 `spaces`.

---
Edited: only `indent` rule is enabled and tweaked, the `react/jsx-indent-props` and `react/jsx-indent` are off

### CODE REVIEW

**Comment by mareklibra** [comment]
Manual change in: pkg/storaged/storage-controls.jsx

File: pkg/apps/application-list.jsx [hunk 1]
```diff
@@ -115,9 +115,9 @@ class ApplicationList extends React.Component {
         function refresh() {
             var config = cockpit.manifests["apps"].config || { };
             PackageKit.refresh(self.props.metainfo_db.origin_files,
-                               config.appstream_config_packages || [ ],
-                               config.appstream_data_packages || [ ],
-                               data => self.setState({ progress: data })).
+                config.appstream_config_packages || [ ],
+                config.appstream_data_packages || [ ],
+                data => self.setState({ progress: data })).
                 finally(() => self.setState({ progress: false })).
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply pkg/apps/application-list.jsx]: right, chaining of the `finally` is harder to notice.
We can play with the number of indentation spaces for different rules.

Or maybe better: I can split the indentation rules to separate PRs. In general, we have to have indentation enforced, at least partially. The indentation rules have many options to set, I believe we can get to something reasonable.

File: pkg/docker/storage.jsx [hunk 2]
```diff
@@ -350,18 +350,18 @@
...
-                                                    { render_drive_rows() }
-                                                </table>
-                                            </div>),
-                                      },
-                                      { 'actions': [ { 'caption': _("Reformat and add disks"),
-                                                       'clicked': add_drives,
-                                                       'style': "danger" } ]
-                                      });
+            'body': (
+                <div className="modal-body">
```
Comment (mareklibra) [inline pkg/docker/storage.jsx]: With respect to the comment above, this re-formatting looks good to me.

File: pkg/lib/cockpit-components-dialog.jsx [hunk 3]
```diff
@@ -204,7 +204,7 @@ var DialogFooter = React.createClass({
                 <button
                     className={ cancel_style }
                     onClick={ this.cancel_click.bind(this) }
-                    >{ cancel_caption }</button>
+                >{ cancel_caption }</button>
```
Comment (mareklibra) [inline pkg/lib/cockpit-components-dialog.jsx]: This needs manual reformatting - move `>` to upper line

File: pkg/packagekit/updates.jsx [hunk 4]
```diff
@@ -468,7 +468,7 @@ class ApplyUpdates extends React.Component {
             actionHTML = (
                 <span>
                     <strong>{ PK_STATUS_STRINGS[lastAction.status] || PK_STATUS_STRINGS[PK.Enum.STATUS_UPDATE] }</strong>
-                    &nbsp;{lastAction.package}
+                    {lastAction.package}
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply pkg/packagekit/updates.jsx]: Yes, removed automatically by `eslint:fix`.
The `&nbsp;` is anti-patern, IMO.
I can grep for it prior applying indentation rules and replace all occurrences by CSS `padding`.

File: pkg/apps/application.jsx [hunk 5]
```diff
@@ -63,10 +63,10 @@ class Application extends React.Component {
             return urls.map(url => {
                 if (url.type == 'homepage') {
                     return (<div className="app-links">
-                                <a href={url.link} target="_blank" rel="noopener" data-linkedhost={url.link}>
+                        <a href={url.link} target="_blank" rel="noopener" data-linkedhost={url.link}>
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply pkg/apps/application.jsx]: I agree, that's what I tried to say with my proposal above.

**Comment by mareklibra** [comment]
Only `indent` rule is enabled now.

Remaining `react/jsx-indent-props` and `react/jsx-indent` are `off` which seems reasonable to me.

File: pkg/storaged/vgroups-panel.jsx [hunk 6]
```diff
@@ -96,7 +96,8 @@ export class VGroupsPanel extends React.Component {
             );
         }
 
-        var vgroups = Object.keys(client.vgroups).sort(cmp_vgroup).map(make_vgroup);
+        var vgroups = Object.keys(client.vgroups).sort(cmp_vgroup)
+                .map(make_vgroup);
```
Comment (martinpitt) [inline pkg/storaged/vgroups-panel.jsx]: Not sure if that really makes things easier to read, but I don't object.

**Comment by martinpitt** [comment]
The jshint indention changes collided with #9034 which removed them entirely, so this needs a rebase.

**Pull request closed without merging**